bin/test-cmssw.sh

```diff
@@ -32,107 +34,154 @@ if [ "${ARCHS}" = "" ] ; then
-  fi
-  echo "Found release: ${cmssw_ver}"
-  INST_OPTS="-a $SCRAM_ARCH --debug install --ignore-size --jobs 2 -y"
-  $WORKSPACE/inst/$SCRAM_ARCH/common/cmspkg $INST_OPTS cms+cmssw+${cmssw_ver}
-  INSTALL_PACKAGES=""
-  for pkg in gcc-fixincludes cms+dasgoclient ; do
-    pkg=$(($WORKSPACE/inst/$SCRAM_ARCH/common/cmspkg -a $SCRAM_ARCH search "$pkg" | grep 'CMS Experiment package' | tail -1 | sed 's| .*||') || true)
-    if [ "${pkg}" != "" ] ; then
-      INSTALL_PACKAGES="${INSTALL_PACKAGES} $pkg"
+  if $BUILDTIME ; then
+    cd $WORKSPACE/inst
+    echo ${SCRAM_ARCH} >> $WORKSPACE/res.txt
+    if [ $(echo ${INVALID_ARCHS} | tr ' ' '\n' | grep "^${arch}$" | wc -l) -gt 0 ] ; then
+      echo ${SCRAM_ARCH}.SKIP >> $WORKSPACE/res.txt
+      continue
     fi
-  done
-  if [ "${INSTALL_PACKAGES}" != "" ] ; then
-    $WORKSPACE/inst/$SCRAM_ARCH/common/cmspkg $INST_OPTS ${INSTALL_PACKAGES}
-  fi
-  export cmssw_ver
-  (
+    [ "${parch}" != "" ] && rm -rf ${parch}
+    parch="${arch}"
+    rm -rf ./$SCRAM_ARCH ; mkdir -p ./$SCRAM_ARCH
+    cd ./$SCRAM_ARCH
+    touch cmssw.rel
+    $(source /cvmfs/cms.cern.ch/cmsset_default.sh >/dev/null 2>&1; scram -a $SCRAM_ARCH list -c CMSSW | grep -v '/cmssw-patch/' | grep ' CMSSW_' >cmssw.rel) || true
+    cat cmssw.rel
+    cmssw_ver=""
+    boot_repo="cms"
+    for v in $(grep ${RELEASE_INST_DIR}/ cmssw.rel | grep '_[0-9][0-9]*_X_' | awk '{print $3}' | tac) ; do
+      if [ -e $v/build-errors ] ; then continue ; fi
+      cmssw_ver=$(basename $v)
+      boot_repo=cms.$(echo $v | cut -d/ -f4)
+      if [ "${boot_repo}" = "cms.sw" ]; then boot_repo=cms.$(echo $v | cut -d/ -f6); fi
+      break
+    done
+    if [ "${cmssw_ver}" = "" ] ; then
+      cmssw_ver=$(grep /cvmfs/cms.cern.ch/ cmssw.rel | tail -1 | awk '{print $2}' || true)
+    fi
+    if ! sh -ex $WORKSPACE/inst/bootstrap.sh -server ${CMSREP} -r ${boot_repo} -a $SCRAM_ARCH setup ; then
+      echo ${SCRAM_ARCH}.BOOT.ERR >> $WORKSPACE/res.txt
+      continue
+    fi
+    echo ${SCRAM_ARCH}.BOOT.OK >> $WORKSPACE/res.txt
+    if [ "${cmssw_ver}" = "" ] ; then
+      echo "Warnings: No CMSSW version available for $SCRAM_ARCH"
+      continue
+    fi
+    echo "Found release: ${cmssw_ver}"
+
+    INST_OPTS="-a $SCRAM_ARCH --debug install --ignore-size --jobs 2 -y"
+    $WORKSPACE/inst/$SCRAM_ARCH/common/cmspkg $INST_OPTS cms+cmssw+${cmssw_ver}
+    INSTALL_PACKAGES=""
+    for pkg in gcc-fixincludes cms+dasgoclient ; do
+      pkg=$(($WORKSPACE/inst/$SCRAM_ARCH/common/cmspkg -a $SCRAM_ARCH search "$pkg" | grep 'CMS Experiment package' | tail -1 | sed 's| .*||') || true)
+      if [ "${pkg}" != "" ] ; then
+        INSTALL_PACKAGES="${INSTALL_PACKAGES} $pkg"
+      fi
+    done
+    if [ "${INSTALL_PACKAGES}" != "" ] ; then
+      $WORKSPACE/inst/$SCRAM_ARCH/common/cmspkg $INST_OPTS ${INSTALL_PACKAGES}
+    fi
+     export cmssw_ver
+    (
+      export CMS_PATH=/cvmfs/cms-ib.cern.ch
+      export SITECONFIG_PATH=/cvmfs/cms-ib.cern.ch/SITECONF/local
```

Comment:
these two lines are common for all archs/releases and containers. So you can move these above the arch loop